### PR TITLE
Deduplicate list of directories

### LINKED ISSUE #539: [Plugin] [1.12] plugin binaries are overshadowed by themselves if they appear in PATH twice
**Kubernetes version** (use `kubectl version`): 1.12.0-beta.2

**What happened**:

I have added a directory that contains my plugins to my $PATH twice, like:

```
export PATH="$HOME/.krew/bin:$PATH:$HOME/.krew/bin"
```

then I ran `kubectl plugin list` only to find out the plugins are overshadowed by themselves, e.g.

```
[...]
/Users/ahmetb/.krew/bin/kubectl-open-svc
  - warning: /Users/ahmetb/.krew/bin/kubectl-open-svc is overshadowed by a similarly named plugin: /Users/ahmetb/.krew/bin/kubectl-open-svc
```
(not the paths are exact)

Also `kubectl open-svc` doesn't work because of this now.

**What you expected to happen**:

Deduplication of exact path names.

**How to reproduce it** (as minimally and precisely as possible):

1. Create an executable `kubectl-foo` in a dir
2. add that dir to $PATH twice
3. run `kubectl plugin list`.
4. Observe plugins are overshadowed by exactly themselves.

/cc @soltysh
^ apologies if this is not the correct person to cc.
/cc @lbb

### PR DESCRIPTION
**Pull request by soltysh**
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes/kubectl/issues/539

/assign @juanvallejo 

Btw. we should add some tests for plugin command.

**Release note**:

```release-note
Deduplicate PATH items when reading plugins.
```

### CODE REVIEW

File: pkg/kubectl/cmd/plugin.go [hunk 1]
```diff
@@ -227,3 +228,15 @@ func isExecutable(fullPath string) (bool, error) {
 
 	return false, nil
 }
+
+func unique(items []string) []string {
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/plugin.go]: We could alternatively use the "sets" package from apimachinery: `"k8s.io/apimachinery/pkg/util/sets.NewString"`

**Comment by soltysh** [comment]
@juanvallejo good point, updated. PTAL

**Review by juanvallejo — APPROVED** [review]

**Comment by ahmetb** [comment]
Should this be cherry-picked into 1.12?

**Comment by juanvallejo** [comment]
> Should this be cherry-picked into 1.12?

Yes, this should be part of the 1.12 release

**Comment by ahmetb** [comment]
Please add the cherrypick-candidate lable.

**Comment by soltysh** [comment]
> Please add the cherrypick-candidate label.

It was removed since nobody used it. Only cherrypick-approved is useful on backport PRs.

**Comment by ahmetb** [comment]
I see. So potentially this is delayed until 1.13 or it could go out in a patch release for 1.12?

**Comment by soltysh** [comment]
https://github.com/kubernetes/kubernetes/pull/69170 was a 1.12 cherry-pick, so watch out for release patch.

**Pull request merged**